### ISSUE TITLE
Add floating point operators/conversions for Fw::Time

### ISSUE THREAD

**Issue by timcanham** [nasa/fprime#1088]
| | |
|:---|:---|
|**_F´ Version_**| any |
|**_Affected Component_**| all |
---
## Feature Description

Add `F32`/`F64` time conversion and math operators.

## Rationale

Sometimes a user with do some time math in floating point and then switch top `Fw::Time`. It would be nice to add conversions and operators that include floating point.

For example:

```
Fw::Time myTime = 2.5;

myTime += 6.2;

F32 newTime = myTime;

```

**Comment by metzdigital** [comment]
I recommend using only F64. There were issues with the profiler on OWLS because we use a TB_WORK_STATION time base and so the finest time step we could resolve was in the minutes.

**Comment by LeStarch** [comment]
Yes, linux epoch w/ sub seconds does not fit into an F32 without precision loss.

**Comment by matt392code** [comment]
[time-operators.txt](https://github.com/user-attachments/files/18304113/time-operators.txt)

Proposed implementation floating point operators for Fw::Time with the following considerations:
1. Focused on F64 support only since F32 would lead to precision loss, especially with Linux epoch timestamps and sub-seconds.

2. The implementation includes:
   - Conversion constructor from F64 to Time
   - Conversion operator from Time to F64
   - Addition and subtraction operators with F64 values
   - Both member operators (+=, -=) and free-standing operators (+, -)

3. The conversion logic:
   - When converting from F64, splits the value into whole seconds and microseconds
   - When converting to F64, combines seconds and microseconds into a single floating-point value

4. The implementation maintains microsecond precision while handling the conversions.
Key features of this implementation:
- Uses microsecond precision (1e-6 seconds) to maintain compatibility with existing F-Prime time handling
- Handles floating-point math correctly for both positive and negative values
- Provides intuitive operator syntax matching the examples in the ticket

To use this implementation:
```cpp
Fw::Time myTime = 2.5;      // Creates time of 2 seconds, 500000 microseconds
myTime += 6.2;              // Adds 6.2 seconds
F64 newTime = myTime;       // Converts back to floating point (would be 8.7)
```
